glean-core/csharp/Glean/Dispatchers.cs

```diff
@@ -3,24 +3,213 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
 
 namespace Mozilla.Glean
 {
     internal class Dispatchers
```

Comment:
I see you're making this `static` heavy. No concern on this, just a different paradigm with respect to the Lazy patterns used elsewhere :)

Reply:
I was also a little worried about this, but the option would be to use a Singleton pattern like the base Glean object.  If static ends up becoming a problem for us, I'm okay with changing it out.